cgi-bin/CATS/Problem/Storage.pm

```diff
@@ -171,6 +179,21 @@ sub change_file {
     );
 }
 
+sub delete_file {
+    my ($self, $cid, $pid, $file, $dir, $message) = @_;
+    $user->{git_author_name} && $user->{git_author_email} or return (-1, msg(1167));
+
+    my $repo = get_repo($pid);
+    _is_file_exist($repo, $file, $dir) or return (-1, msg(1206, $file));
+
+    $repo->rm(File::Spec->catfile($repo->get_dir, $file));
```

Comment:
1) `$dir` should be used here
2) `$repo->get_dir` should not be used here
3) It is inconsitent to pass file and dir separately here, but together in `replace_file_content` above